rasa/core/training/dsl.py

```diff
@@ -197,6 +197,40 @@ def __init__(self, domain, interpreter, template_vars=None, use_e2e=False):
 
         return story_steps
 
+    def verify_stories_format(self, filename):
+        stories = self._remove_comments(filename)
+
+        for line in range(len(stories)):
+            stories[line] = stories[line].strip()
+
+        for line in stories:
+            if len(line) > 0 and line[0] not in {"*", "#", "-", ">"}:
+                logger.error(
+                    "There is an error with the following line "
+                    "in your stories file"
+                    " {}:".format(filename)
+                )
+                logger.error(line)
+        return stories
+
+    def _clean_string(self, st: Text, story_file: Text):
+        fragments = []
+        while st:
+            fragment, open_, st = st.partition("<!--")
+            _, close, st = st.partition("-->")
+            if open_ and not close or close and not open_:
+                logger.error(
+                    "The story file {} has a unclosed comment".format(story_file)
+                )
+            fragments.append(fragment)
+        return "".join(fragments)
```

Comment:
I'm really unsure what's happening here, can you give it a method description and better variables please?